pallets/protos/src/lib.rs

```diff
@@ -535,6 +538,46 @@ pub mod pallet {
 			<ProtosByTag<T>>::get(&tags)
 		}
 
+		fn is_proto_having_any_tags(proto_hash: &Hash256, tags: &Vec<Tags>) -> bool {
+			if let Some(struct_proto) = <Protos<T>>::get(proto_hash) {
+				tags.into_iter().any(|tag| struct_proto.tags.contains(&tag))
+			} else {
+				false
+			}
+		}
+
+		pub fn get_by_tags(tags: Vec<Tags>, owner: Option<T::AccountId>, limit: u32) -> Option<Vec<Hash256>> {
+
+			match owner {
+				Some(owner) => {
+					if let Some(vector_protos) = <ProtosByOwner<T>>::get(ProtoOwner::User(owner)) {
+						let iter_protos = vector_protos.into_iter();
+						let iter_protos_filtered = iter_protos.filter(|proto| Self::is_proto_having_any_tags(proto, &tags));
```

Comment:
does this returns another iterator? Must make sure about it or it will need to go thru ALL

Reply:
Yes, this one returns an iterator

Reply:
You are expecting `<ProtosByOwner<T>>::get(ProtoOwner::User(owner))` to return an iterator, correct?

Reply:
`get` is no good, we need to change the type like you hinted the other day into ITERABLE
https://docs.substrate.io/rustdocs/latest/frame_support/storage/trait.IterableStorageMap.html

Reply:
Got it. Thanks